src/main/java/de/unistuttgart/iaas/amyassist/amy/core/speech/SpeechRecognizer.java

```diff
@@ -0,0 +1,230 @@
+ *
+ */
+package de.unistuttgart.iaas.amyassist.amy.core.speech;
+
+import java.io.IOException;
+
+import javax.sound.sampled.AudioFormat;
+import javax.sound.sampled.AudioInputStream;
+import javax.sound.sampled.AudioSystem;
+import javax.sound.sampled.LineUnavailableException;
+import javax.sound.sampled.TargetDataLine;
+
+import edu.cmu.sphinx.api.Configuration;
+import edu.cmu.sphinx.api.SpeechResult;
+import edu.cmu.sphinx.api.StreamSpeechRecognizer;
+
+/**
+ * Class that translate Aduio-Input into Strings, 
+ * powered by CMU Sphinx - https://cmusphinx.github.io/
+ * which is Licenced under BSD
+ * 
+ * @author Kai Menzel
+ */
+public class SpeechRecognizer implements SpeechIO {
+	
+	//wakeup-sleep-shutdown-commands
+	private String wakeUp;
+	private String goSleep;
+	private String shutDown;
+	
+	//Audio Input Source for the Recognition
+	static AudioInputStream ais = null;
+	
+	//Handler who use the translated String for commanding the System
+	SpeechInputHandler inputHandler;
+	SpeechCommandHandler commandHandler = new SpeechCommandHandler();
+	
+
+	//The Recognizer which Handles the Recognition
+	private StreamSpeechRecognizer recognizer;
+	
+	//-----------------------------------------------------------------------------------------------	
+
+	/**
+	 * Creates the Recognizers and Configures them
+	 * @param wakeUp call to start the recognition
+	 * @param goSleep call to stop the recognition
+	 * @param shutDown call to shutdown the whole system
+	 * @param grammarPath Path to the folder of the different Grammars
+	 * @param grammarName Name of the to Used Grammar (without ".gram")
+	 * @param ais set custom AudioInputStream. Set *null* for default microphone input
+	 */
+	public SpeechRecognizer(String wakeUp, String goSleep, String shutDown, String grammarPath, String grammarName, AudioInputStream ais) {
+		this.wakeUp = wakeUp;
+		this.goSleep = goSleep;
+		this.shutDown = shutDown;
+		SpeechRecognizer.ais = ais;
+		
+		//Create the Recognizers
+		try {			
+			SpeechRecognizer.this.recognizer = new StreamSpeechRecognizer(createConfiguration(grammarPath, grammarName));
+		} catch (IOException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+	}
+	
+	//-----------------------------------------------------------------------------------------------
+
+	/**
+	 * @see java.lang.Runnable#run()
+	 * Starts and runs the recognizer
+	 * calls makeDecision() with the recognized String
+	 */
+	@Override
+	public void run() {
+		// TODO Auto-generated method stub
+
+		//check and start audioInputstream
+		while(SpeechRecognizer.ais == null){
+			startInput();
+			Thread.yield();
+		}
+		
+		//starts Recognition
+		this.recognizer.startRecognition(SpeechRecognizer.ais);
+		
+		//Boolean to check if we are supposed to listen (sleeping)
+		boolean listening = false;
+		
+		//The result of the Recognition
+		String speechRecognitionResult;
+		
+		while(true){
```

Comment:
gibt es keinen callback, den man benutzten kann, statt immer `getResult()` aufzurufen?

Reply:
getResult() wartet auf ein ergebnis und gibt es dann zurück, Tim meinte beim testen mit dem StreamRecognizer jedoch dass es ab und an vorkommen kann dass er *null* als rückgabe hat, daher steht er in einer schleife.